pkg/extensions/tbtc/tbtc.go

```diff
@@ -91,18 +96,51 @@ func (t *tbtc) monitorRetrievePubKey(
 	actBackoffFn backoffFn,
 	timeout time.Duration,
 ) error {
+	initialDepositState := chain.AwaitingSignerSetup
+
+	monitoringStartFn := func(
+		handler depositEventHandler,
+	) (subscription.EventSubscription, error) {
+		return t.chain.OnDepositCreated(handler)
+	}
+
+	monitoringStopFn := func(
+		handler depositEventHandler,
+	) (subscription.EventSubscription, error) {
+		return t.chain.OnDepositRegisteredPubkey(func(depositAddress string) {
+			if t.waitDepositStateChangeConfirmation(
```

Comment:
Should we emit a warning if this is not true? I agree that error is good for `actFn`, here error could be too much but warning feels good.